src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -1055,82 +1055,94 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_WriteThreadNetw
     switch (attributeId)
     {
     case ThreadNetworkDiagnostics::Attributes::Channel::Id: {
-        uint16_t channel = otLinkGetChannel(mOTInst);
-        err              = encoder.Encode(channel);
+        err = encoder.EncodeNullableOnCondition(mIsAttached, [this](const auto pAttrEncoder) -> CHIP_ERROR {
```

Comment:
Are you sure this should return null when the Thread device is not attached? The spec only says: "A value of null
SHALL indicate that the Thread interface is not currently configured or operational." and I would interpret that as "Thread role is 'disabled'". Technically, it shouldn't matter for most attributes as anyway, to read them the device must be attached or the device being commissioned over BLE, but I can imagine some commissioners might want to read `RoutingRole` over BLE to learn if the device is a REED or MED etc.

Reply:
>  Thread interface is not currently configured or operational.

 Doesn't Thread role being disable fall into the category "not operation"? To me when thread is not in an attached mode, it is not operational.

Reply:
Thanks, looking good now!